src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.epam.lowcost.DAO;
+
+import com.epam.lowcost.model.Flight;
+import com.epam.lowcost.model.Plane;
+import com.epam.lowcost.util.DateFormatter;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlightDAOImpl implements FlightDAO {
+    private DataSource dataSource;
+    private PlaneDAO planeDAO;
+
+    public FlightDAOImpl(DataSource dataSource, PlaneDAO planeDAO) {
+        this.dataSource = dataSource;
+        this.planeDAO = planeDAO;
```

Comment:
it is bad practice. Usually Service has another service and uses it and then created whole entity. Is it clear ?

Reply:
changed to good practice